pkg/cmd/dev/builder.go

```diff
@@ -41,12 +41,18 @@ func makeBuilderCmd(runE func(cmd *cobra.Command, args []string) error) *cobra.C
 func (d *dev) builder(cmd *cobra.Command, extraArgs []string) error {
 	ctx := cmd.Context()
 	volume := mustGetFlagString(cmd, volumeFlag)
-	args, err := d.getDockerRunArgs(ctx, volume, false)
+	var tty bool
+	if len(extraArgs) == 0 {
+		tty = true
+	}
+	args, err := d.getDockerRunArgs(ctx, volume, tty)
 	args = append(args, extraArgs...)
 	if err != nil {
 		return err
 	}
-	logCommand("docker", args...)
+	if tty {
+		logCommand("docker", args...)
```

Comment:
As I was playing with it I thought it would be more convenient this way. 🤷 With it this way you can do `VAR=$(dev builder echo hi)` and just get `VAR=hi` rather than having a bunch of garbage prepended to the string. I figure at some point someone will do a `dev builder bazel query ...` and they'll be happy they don't have to filter out the logged `docker` command.

Meanwhile if `tty` is true, the assumption is the command is interactive so it makes sense to log the command.